### ISSUE TITLE
Bug with nested labeled for-of loops

### ISSUE THREAD

**Issue by humanchimp** [babel/babel#996]
This is pretty similar to one of the other bugs I presented recently. The difference here is the nesting:

```javascript
({ *a() { b: for (let c of d()) { for (let e of f()) { continue b } } }}) 
```
```
unknown Statement of type "ForOfStatement"
```
[Here](https://babeljs.io/repl/#?experimental=true&playground=false&evaluate=true&loose=false&spec=false&code=(%7B%20*a()%20%7B%20b%3A%20for%20(let%20c%20of%20d())%20%7B%20for(let%20e%20of%20f())%20%7B%20continue%20b%20%7D%20%7D%20%7D%7D)) is a link to the babel playground.
[Here](https://google.github.io/traceur-compiler/demo/repl.html#(%7B%20*a()%20%7B%20b%3A%20for%20(let%20c%20of%20d())%20%7B%20for(let%20e%20of%20f())%20%7B%20continue%20b%20%7D%20%7D%20%7D%7D)%20) is the same thing compiled by traceur.

**Comment by sebmck** [comment]
Looks like a regenerator bug.

**Comment by sebmck** [comment]
Actually it's both.

**Comment by sebmck** [comment]
Labelled `for...of`s cause the parent node to be replaced which breaks the inner `for...of` traversal.

**Comment by neVERberleRfellerER** [comment]
@sebmck 
Can't this be fixed by adding `return loop.body` [after this line](https://github.com/babel/babel/blob/master/src/babel/transformation/transformers/es6/for-of.js#L36) or by moving `return build.node` out of `else` branch?

**Comment by humanchimp** [comment]
removing the return statement from the else clause allows it to compile. It's hard for me to tell if the output is correct tho, sorry about that.

**Comment by sebmck** [comment]
Incredibly sorry for the delay, fixed as of 4.7.9. The `return` fix worked but only as a consequence of how traversal is handled, it's hacky but there's no solid way to recurse back into a parent if it's been replaced. I'll look into better ways of handling this.
